views/js/runner/plugins/navigation/limitBackButton.js

```diff
@@ -0,0 +1,174 @@
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Plugin that limits the use of the back button only when there is no responses in the current item.
+ * So, once a user has entered a response to any interaction, the back button will be disabled.
+ * We can check if an interaction is already answered by listening to `responsechange` on item runner.
+ *
+ * @author Ricardo Proença <ricardo@taotesting.com>
+ */
+define([
+    "jquery",
+    "lodash",
+    'taoTests/runner/plugin'
+], function ($, _, pluginFactory) {
+    'use strict';
+
+    var pluginName = "limitBackButton";
+
+    var defaultEventName = "change";
+
+    var itemInteractions = {};
+
+    var customListeners = [
+        {
+            selector: ".widget-numpad",
+            eventName: "click"
+        }
+    ];
+
+    /**
+     * Gets the unique item key identifier for this plugin
+     *
+     * @param {string} itemIdentifier - item runner identifier
+     * @returns {string}
+     */
+    var getItemKey = function getItemKey(itemIdentifier) {
+        return itemIdentifier + "-limit-back-button";
+    };
+
+    /**
+     * Remove all listeners from an item
+     *
+     * @param {Object} testRunner - test runner, needed to find interactions elements
+     */
+    var removeAllListeners = function removeAllListeners(testRunner) {
+        _.forEach(customListeners, function (plugin) {
+            var $element = testRunner.getAreaBroker().getContainer().find(plugin.selector);
+
+            $element.off("." + pluginName);
+        });
+
+        _.forEach(itemInteractions, function (interaction) {
+            var selector = "[data-serial='" + interaction.serial + "']";
+            var $element = testRunner.getAreaBroker().getContentArea().find(selector);
+
+            $element.off("." + pluginName);
+        });
+    };
+
+    /**
+     * Returns the configured plugin
+     */
+    return pluginFactory({
+
+        name: pluginName,
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init: function init() {
+            var testRunner = this.getTestRunner();
+
+            return testRunner.getPluginStore(this.getName()).then(function (store) {
+                var disableState = false;
+                var processingMoveAction = false;
+
+                var toggleBackButton = function toggleBackButton() {
+                    var previous = testRunner.getPlugin("previous");
+
+                    disableState ? previous.disable() : previous.enable();
+                };
+
+                var createListener = function createListener(element, eventName) {
+                    var event = eventName
+                        ? eventName + "." + pluginName
+                        : defaultEventName + "." + pluginName;
+
+                    element.on(event, function () {
+                        testRunner.itemRunner.trigger("responsechange." + pluginName);
```

Comment:
Caution, the namespace should not be used when triggering an event, unless a very good reason.
The meaning of the namespace is different on `trigger`, as it reduces the scope: only the exact namespaced event can be caught, other `responsechange` event listeners will be discarded.
But maybe is it the expected behavior here?

Reply:
Yes, it is the expected behavior here, because I don't want to trigger an event that can be listened outside of this module.